sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -0,0 +1,1050 @@
+new executable instance of the graph. An executable graph cannot be converted
+to a modifiable graph. After finalizing a graph in the modifiable state it is
+valid for a user to add additional nodes and finalize again to create subsequent
+executable graphs. The state of a `command_graph` object is made explicit by
+templating on state to make the class strongly typed, with the default template
+argument being `graph_state::modifiable` to reduce code verbosity on
+construction.
+
+.Graph State Diagram
+image::images/sycl_ext_oneapi_graph/command_graph-state.svg[]
+
+==== Executable Graph Update
+
+A graph in the executable state can have each nodes inputs & outputs updated
+using the `command_graph::update()` method. This takes a graph in the
+modifiable state and updates the executable graph to use the node input &
+outputs of the modifiable graph, a technique called _Whole Graph Update_. The
+modifiable graph must have the same topology as the graph originally used to
+create the executable graphs, with the nodes added in the same order.
+
+==== Graph Member Functions
+
+Table 5. Constructor of the `command_graph` class.
+[cols="2a,a"]
+|===
+|Constructor|Description
+
+|
+[source,c++]
+----
+using namespace ext::oneapi::experimental;
+command_graph(const property_list &propList = {});
+----
+|Creates a SYCL `command_graph` object in the modifiable state.
+Zero or more properties can be provided to the constructed SYCL `command_graph`
+via an instance of `property_list`.
+
+Preconditions:
+
+* This constructor is only available when the `command_graph` state is
+  `graph_state::modifiable`.
+
+Parameters:
+
+* `propList` - Optional parameter for passing properties. No new properties are
+  defined by this extension.
+
+|===
+
+Table 6. Member functions of the `command_graph` class.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source,c++]
+----
+using namespace ext::oneapi::experimental;
+node add(const std::vector<node>& dep = {});
+----
+|This creates an empty node which is associated to no task. Its intended use is
+either a connection point inside a graph between groups of nodes, and can
+significantly reduce the number of edges ( O(n) vs. O(n^2) ). Another use-case
+is building the structure of a graph first and adding tasks later.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::modifiable`.
+
+Parameters:
+
+* `dep` - Nodes the created node will be dependent on.
+
+Returns: The empty node which has been added to the graph.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if a queue is recording
+  commands to the graph.
+
+|
+[source,c++]
+----
+using namespace ext::oneapi::experimental;
+template<typename T>
+node add(T cgf, const std::vector<node>& dep = {});
+----
+|This function adds a command group function object to a graph. The function
+object can contain single or multiple commands such as a host task which is
```

Comment:
What does "multiple commands" refer to here? 

The SYCL specification ([§4.9.3](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:command.group.scope)) allows command-group functions to statically contain more than one command, but only allows a single command to be executed by each command-group function: 
> A [command group scope](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command-group-scope), as defined in [Section 3.7.1](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:executionmodel), may execute a single [command](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command) such as invoking a kernel, copying memory, or executing a host task. It is legal for a [command group scope](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command-group-scope) to statically contain more than one call to a [command](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command) function, but any single execution of the [command group function object](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command-group-function-object) may execute no more than one [command](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command)